pilot/tools/debug/pilot_cli.go

```diff
@@ -114,12 +126,23 @@ func NewPodInfo(nameOrAppLabel string, kubeconfig string) *PodInfo {
 				IP:        pod.Status.PodIP,
 			}
 		}
+		if istio, ok := pod.ObjectMeta.Labels["istio"]; ok && istio == nameOrAppLabel {
+			log.Infof("Found pod %s.%s~%s matching app label %q", pod.Name, pod.Namespace, pod.Status.PodIP, nameOrAppLabel)
+			return &PodInfo{
+				Name:      pod.Name,
+				Namespace: pod.Namespace,
+				IP:        pod.Status.PodIP,
+			}
+		}
 	}
 	log.Warnf("Cannot find pod with name or app label matching %q in registry.", nameOrAppLabel)
 	return nil
 }
 
 func (p PodInfo) makeNodeID() string {
+	if strings.HasPrefix(p.Name, "istio-ingress") {
```

Comment:
https://github.com/istio/istio/pull/5948 has a change for different proxy types. I like this auto detection, though not sure how reliable it could be.

Reply:
Can you also add the case for istio-ingressgateway, which uses `router` prefix.

Reply:
It would be nice to have both to gain some convenience without losing flexibility. 

Added ingressgateway and egressgateway with router prefix. I guess actually we can ignore ingress node type since istio-ingress is not used in v2, but anyway I will keep it here.